backend/ethereum/channel/contractbackend.go

```diff
@@ -130,6 +143,17 @@ func (c *ContractBackend) NewTransactor(ctx context.Context, gasLimit uint64,
 	auth.GasLimit = gasLimit
 	auth.Context = ctx
 
+	nonce, err := c.PendingNonceAt(ctx, acc.Address)
+	if err != nil {
+		return nil, errors.Wrap(err, "fetching nonce")
```

Comment:
I think you have to check for `IsChainNotReachableError` here.

Reply:
Why?

Reply:
@matthiasgeihs  Because, we explicitly inform the user by returning a named error when there are problems connecting with the blockchain node. 

@ggwpez I have fixed this now.

Reply:
I see. It seems a bit unclean that we introduce such specific error handling logic in so many places. We could add wrappers for the handful of functions that need this handling and then consistently use the wrappers, right? It's ok here, but it's something worth discussing in general, I think.

Reply:
Yes the initial approach calls the error wrapping on the lowest level, where we also could have done it on the highest…  
Its much more effort like this i think.